### ISSUE TITLE
The filename and paths are different (worse) in the latest update

### ISSUE THREAD

**Issue by eugenet8k** [cypress-visual-regression/cypress-visual-regression#247]
In 5.0.0 it was (expected):
> /myapp/cypress/snapshots/base/test.cy.js/forgot-password-android.png

as I call command as
```
cy.compareSnapshot('forgot-password-android')
```

Now in 5.0.2:
> /myapp/cypress/snapshots/base/cypress/e2e/test.cy.js/forgot_password_android.png

Extra `/cypress/e2e` path and the file name for some reason converted to snakecase.

**Comment by iacopo-carlini-LEI** [comment]
I am having the same issue after upgrading from 5.0.0 to 5.0.2

**Comment by Mugiwara84** [comment]
Same problem

**Comment by Lazlorian** [comment]
We also had to go back to version 5.0.0 for this reason.

It is expecting a different folder structure for the 'visualRegressionBaseDirectory'. 
To get 5.0.2 running you need to add folders below the base directory: `source/e2e/ + whatever/folders/you/have/spec-file-name.cy.ts`

'src/fixtures/pictures/spec-file.cy.ts/snapshot.png' 
-> becomes
'src/fixtures/pictures/source/e2e/whatever/folder/spec-file.cy.ts/snapshot.png' 


In 'open' mode it also adds a `source/e2e/ + whatever/folders/you/have` to the Cypress screenshot output folder, before the spec file named folder, and in 'run' mode it also adds the spec name again as a folder before all that. 

Seems like a mess, and I don't think it is a good idea to mess with the output folder structure. In our case we even have other dependencies on the output folder.

**Comment by iacopo-carlini-LEI** [comment]
> We also had to go back to version 5.0.0 for this reason.
> 
> It is expecting a different folder structure for the 'visualRegressionBaseDirectory'. To get 5.0.2 running you need to add folders below the base directory: `source/e2e/ + whatever/folders/you/have/spec-file-name.cy.ts`
> 
> `src/fixtures/pictures/spec-file.cy.ts/snapshot.png` -> becomes `src/fixtures/pictures/source/e2e/whatever/folder/spec-file.cy.ts/snapshot.png`
> 
> It also adds a `source/e2e/ + whatever/folders/you/have` to the Cypress screenshot output folder, before the spec file named folder.
> 
> Seems like a mess, and I don't think it is a good idea to mess with the output folder structure. Also having to replicate your existing folder structure for the fixtures just seems like an additional hassle to maintain, although could be beneficial if you have non unique spec file names (**don't!**).

Applying this workaround solved the issue but I definitively don't like my project folder structure for snapshots. Needs a fix

**Comment by abdalla-rko** [comment]
This is still not fixed upgrading to 5.0.3 or 5.2.0 didn't solve the path issue.

**Comment by Zaista** [comment]
TLDR:

- the extra `cypress/e2e` part in the screenshot paths will not be removed

Full history:
There are quite a few problems that handling this issue makes it really annoying:

1. Cypress does not expose the relative path without this added `cypress/e2e` part, nor does it provide any way to determine it ever since it switched to `specPattern`. There's no sense in trying to parse and calculate this, just for the sake of getting rid of this small part in the folder structure. We raised this issue to improve it, and perhaps voting/reacting on it might speed it up: https://github.com/cypress-io/cypress/issues/29048. Once this is fixed from Cypress side, I would gladly make the change.
2. We already tried switching to the other extreme, no nested structure at all, which is an easy solution, just causing the spec name as a folder container for images, but the community did not like that, as seen here #225 and here #238, so in the end we gave up on that approach
3. We also back in the day added this `INTEGRATION_FOLDER` property (PR #139) where you could specify the location of spec files, but at the time it also served another, more important purpose of finding actual screenshots, and since we found a better way to get a path to actual screenshots, maintaining something like that just for the purpose of making a path slightly cleaner does not justify the effort imo
4. Finally, this longer path structure might seem like `more` stuff to maintain, but it really isn't, the choice boiled down to either no nested folder structure at all, or keep the full path, the first option was causing problems to people who use same spec names, so latter option was chosen in the end

Now all of the above is just related to the nasty `cypress/e2e` part in the path name, any other problem should be fixed, if not, please let me know

**Comment by eugenet8k** [comment]
I guess we all stay on v5.0.0. Thanks for the breakdown @Zaista, but I wish there would be flexibility while waiting for the good Cypress API, to pass the path via a temporary func and just remove `cypress/e2e` with string replace. As it's that annoying for many people.
